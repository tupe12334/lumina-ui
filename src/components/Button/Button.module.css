.button {
  display: inline-flex;
  align-items: center;
  justify-content: center;
  border-radius: 6px;
  font-weight: 500;
  transition: all 0.2s ease;
  border: none;
  cursor: pointer;
  font-family: inherit;
}

.button:focus-visible {
  outline: none;
  box-shadow: 0 0 0 2px rgba(59, 130, 246, 0.5);
}

.button:disabled {
  pointer-events: none;
  opacity: 0.5;
}

/* Variants */
.primary {
  background-color: #2563eb;
  color: white;
}

.primary:hover:not(:disabled) {
  background-color: #1d4ed8;
}

.secondary {
  background-color: #f3f4f6;
  color: #111827;
}

.secondary:hover:not(:disabled) {
  background-color: #e5e7eb;
}

.outline {
  border: 1px solid #d1d5db;
  background-color: transparent;
  color: #374151;
}

.outline:hover:not(:disabled) {
  background-color: #f9fafb;
}

/* Sizes */
.sm {
  height: 32px;
  padding: 0 12px;
  font-size: 14px;
}

.md {
  height: 40px;
  padding: 0 16px;
  font-size: 16px;
}

.lg {
  height: 48px;
  padding: 0 24px;
  font-size: 18px;
}