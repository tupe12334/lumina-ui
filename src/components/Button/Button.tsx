import React from 'react'
import { clsx } from 'clsx'
import styles from './Button.module.css'

export interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  variant?: 'primary' | 'secondary' | 'outline'
  size?: 'sm' | 'md' | 'lg'
  children: React.ReactNode
}

export const Button = React.forwardRef<HTMLButtonElement, ButtonProps>(
  (props, ref) => {
    const { className, variant, size, children, ...otherProps } = props
    const finalVariant = variant !== undefined ? variant : 'primary'
    const finalSize = size !== undefined ? size : 'md'
    return (
      <button
        className={clsx(
          styles.button,
          styles[finalVariant],
          styles[finalSize],
          className
        )}
        ref={ref}
        {...otherProps}
      >
        {children}
      </button>
    )
  }
)

Button.displayName = 'Button'