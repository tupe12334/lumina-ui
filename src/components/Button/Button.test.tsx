import { render } from '@testing-library/react'
import { test } from 'vitest'
import { Button } from './Button'

test('Button renders correctly with default props', () => {
  const { container } = render(<Button>Click me</Button>)
  expect(container.firstChild).toMatchSnapshot()
})

test('Button renders correctly with primary variant', () => {
  const { container } = render(<Button variant="primary">Primary Button</Button>)
  expect(container.firstChild).toMatchSnapshot()
})

test('Button renders correctly with secondary variant', () => {
  const { container } = render(<Button variant="secondary">Secondary Button</Button>)
  expect(container.firstChild).toMatchSnapshot()
})

test('Button renders correctly with outline variant', () => {
  const { container } = render(<Button variant="outline">Outline Button</Button>)
  expect(container.firstChild).toMatchSnapshot()
})

test('Button renders correctly with small size', () => {
  const { container } = render(<Button size="sm">Small Button</Button>)
  expect(container.firstChild).toMatchSnapshot()
})

test('Button renders correctly with medium size', () => {
  const { container } = render(<Button size="md">Medium Button</Button>)
  expect(container.firstChild).toMatchSnapshot()
})

test('Button renders correctly with large size', () => {
  const { container } = render(<Button size="lg">Large Button</Button>)
  expect(container.firstChild).toMatchSnapshot()
})

test('Button renders correctly when disabled', () => {
  const { container } = render(<Button disabled>Disabled Button</Button>)
  expect(container.firstChild).toMatchSnapshot()
})

test('Button renders correctly with custom className', () => {
  const { container } = render(<Button className="custom-class">Custom Button</Button>)
  expect(container.firstChild).toMatchSnapshot()
})

test('Button renders correctly with all props combined', () => {
  const { container } = render(
    <Button
      variant="outline"
      size="lg"
      disabled
      className="custom-class"
      data-testid="complex-button"
    >
      Complex Button
    </Button>
  )
  expect(container.firstChild).toMatchSnapshot()
})