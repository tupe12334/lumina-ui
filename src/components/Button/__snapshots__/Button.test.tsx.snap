// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button renders correctly when disabled 1`] = `
<button
  class="_button_2a1efe _primary_2a1efe _md_2a1efe"
  disabled=""
>
  Disabled Button
</button>
`;

exports[`Button renders correctly with all props combined 1`] = `
<button
  class="_button_2a1efe _outline_2a1efe _lg_2a1efe custom-class"
  data-testid="complex-button"
  disabled=""
>
  Complex Button
</button>
`;

exports[`Button renders correctly with custom className 1`] = `
<button
  class="_button_2a1efe _primary_2a1efe _md_2a1efe custom-class"
>
  Custom Button
</button>
`;

exports[`Button renders correctly with default props 1`] = `
<button
  class="_button_2a1efe _primary_2a1efe _md_2a1efe"
>
  Click me
</button>
`;

exports[`Button renders correctly with large size 1`] = `
<button
  class="_button_2a1efe _primary_2a1efe _lg_2a1efe"
>
  Large Button
</button>
`;

exports[`Button renders correctly with medium size 1`] = `
<button
  class="_button_2a1efe _primary_2a1efe _md_2a1efe"
>
  Medium Button
</button>
`;

exports[`Button renders correctly with outline variant 1`] = `
<button
  class="_button_2a1efe _outline_2a1efe _md_2a1efe"
>
  Outline Button
</button>
`;

exports[`Button renders correctly with primary variant 1`] = `
<button
  class="_button_2a1efe _primary_2a1efe _md_2a1efe"
>
  Primary Button
</button>
`;

exports[`Button renders correctly with secondary variant 1`] = `
<button
  class="_button_2a1efe _secondary_2a1efe _md_2a1efe"
>
  Secondary Button
</button>
`;

exports[`Button renders correctly with small size 1`] = `
<button
  class="_button_2a1efe _primary_2a1efe _sm_2a1efe"
>
  Small Button
</button>
`;
